holidays/calendars/islamic.py

```
@@ -4264,3 +4267,83 @@ def tasua_dates(self, year: int) -> Iterable[tuple[date, bool]]:
+        moon_lat = math.radians(5.128 * math.sin(moon_argument_latitude))
+
+        return sun_lon, moon_lon, moon_lat
+
+    def _check_mabims_criteria(self, check_date: date) -> bool:
+        """Evaluates if the crescent moon meets thresholds at local reference sunset."""
+        j2000_epoch = date(2000, 1, 1)
+        days_since_j2000 = (check_date - j2000_epoch).days + 0.46 
+
+        sun_lon, moon_lon, moon_lat = self._get_solar_lunar_positions(days_since_j2000)
+
+        cos_elongation = max(-1.0, min(1.0, math.cos(moon_lat) * math.cos(moon_lon - sun_lon)))
```

Ruff (0.15.15)

[error] Undefined name `math`

(F821)